### PR TITLE
[1.20.1] Show unmet growth conditions in Magnifying Glass tooltip

### PR DESCRIPTION
**Pull request by unilock**
yeah

### CODE REVIEW

File: common/src/main/java/com/agricraft/agricraft/common/block/entity/CropBlockEntity.java [hunk 1]
```diff
@@ -504,6 +504,18 @@ public void addMagnifyingTooltip(List<Component> tooltip, boolean isPlayerSneaki
...
 			tooltip.add(Component.literal("  ").append(Component.translatable("agricraft.tooltip.magnifying.requirement." + (response.isLethal() ? "lethal" : response.isFertile() ? "fertile" : "not_fertile"))));
+			// crop conditions
+			if (!response.isFertile()) {
+				if (!this.checkGrowthSpace(this.plant.getPlantHeight(this.growthStage))) {
+					tooltip.add(Component.literal("  - ").append(Component.translatable("agricraft.tooltip.magnifying.condition.growth_space")));
+				}
+				int strength = this.genome.getStatGene(AgriStatRegistry.getInstance().strengthStat()).getTrait();
+				AgriGrowthConditionRegistry.getInstance().stream().forEach(condition -> {
+					if (!condition.check(this, this.level, this.getBlockPos(), strength).isFertile()) {
+						tooltip.add(Component.literal("  - ").append(Component.translatable("agricraft.tooltip.magnifying.condition."+condition.getId())));
```
Comment (InfinityRaider) [inline common/src/main/java/com/agricraft/agricraft/common/block/entity/CropBlockEntity.java:515]: GrowConditions have a method to provide a tooltip with feedback: 
https://github.com/AgriCraft/AgriCraft/blob/master/src/main/java/com/infinityraider/agricraft/api/v1/requirement/IAgriGrowCondition.java#L47

Using this is preferable.

Comment (unilock) [reply common/src/main/java/com/agricraft/agricraft/common/block/entity/CropBlockEntity.java:515]: Not in 1.20.1 they don't :p

Comment (InfinityRaider) [reply common/src/main/java/com/agricraft/agricraft/common/block/entity/CropBlockEntity.java:515]: Ah I see, my apologies.
In that case ignore me, one of these days I'll set up a dev environment for this again.
Thanks for your contribution.

**Comment by Ketheroth** [comment]
I'm closing you're pr as I implemented it in f5483851ac1060e0292a442b573977d2ee244d93 and I added back the method to provide the tooltips like InfinityRaider said.
I added you as a co-author as I took some of your code and didn't want your work to go to waste

**Pull request closed without merging**